### ISSUE TITLE
Intermitent crashes on swscaler

### ISSUE THREAD

**Issue by robertoandrade** [bytedeco/javacv#146]
I often times get a VM crash when trying to use `FfmpegFrameGrabber` to decode an H264 stream composed by an iOS device (after AVC converting it to Annex B H264). 

Some times I see crashes when reaching the end of the stream and trying to stop the grabber simultaneously (which I'm not sure is actually related to `swscaler` but will post more details as soon as I see it happen again), but the one I'm getting somewhat consistently is both a warning message and then the crash when grabbing (again, the crash only happens some times so wondering what I could be doing or what could be causing it):

```
[swscaler @ 0xc54eb600] deprecated pixel format used, make sure you did set range correctly
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0xc25cf0b6, pid=6, tid=3329616704
#
# JRE version: OpenJDK Runtime Environment (7.0_79-b14) (build 1.7.0_79-b14)
# Java VM: OpenJDK Server VM (24.79-b02 mixed mode linux-x86 )
# Derivative: IcedTea 2.5.5
# Distribution: Debian GNU/Linux 8.0 (jessie), package 7u79-2.5.5-1~deb8u1
# Problematic frame:
# C  [libswscale.so.3+0x660b6]
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /app/hs_err_pid6.log
#
# If you would like to submit a bug report, please include
# instructions on how to reproduce the bug and visit:
#   http://icedtea.classpath.org/bugzilla
#
```

**Comment by saudet** [comment]
Does this happen with ffplay as well?

**Comment by saudet** [comment]
Version 1.0 comes with a newer version of FFmpeg. Does it still happen with version 1.0?

**Comment by robertoandrade** [comment]
I'll give it a try and let you know

**Comment by saudet** [comment]
Given the lack of feedback, I'm guessing everything works fine with JavaCV 1.0, but let me know if this isn't the case. Thanks!

**Comment by itismadhan** [comment]
This is happening in 1.1; it happens when the iOS device orientation changes from portrait to landscape or vice versa.

iOS version - 9.3.2

**Comment by robertoandrade** [comment]
Yeah, I had to workaround that by creating my own implementation of
FFmpegFrameGrabber that constantly inspects the input frame dimension and
reinitialize the private buffers on the parent class.

**Comment by saudet** [comment]
So this is a duplicate of #77? @robertoandrade It would be great if you could send a pull request of your modifications so we could merge this in!

**Comment by robertoandrade** [comment]
It is, but had to rework the private state manipulation a bit after updating to latest (1.1), I no longer work for the company where I had done such change so no longer have access to the source but will try to check back and see what the latest state reset logic looked like.

**Comment by ps20lee** [comment]
The workaround that Roberto did actually works great when we tried using various iphones.
 we tried using  Iphone 6 plus, however, we still get jvm crash.. still investigating..may be this caused by only iphone 6 plus..

I see many "invalid image channel: 5"...
